gpt_index/indices/query/base.py

```diff
@@ -129,6 +131,18 @@ def _should_use_node(self, node: Node) -> bool:
                 if w in words:
                     return False
 
+        if (
```

Comment:
ubernit: i'm not a fan of parentheses nesting in if statements, could we do something like:
```
sim_cutoff_exists = (
  similarity_tracker is not None and similarity_tracker.similarity_cutoff is not None
)
if sim_cutoff_exists:
  ....
```
same with comparing similarities below. Mostly a stylistic choice but something I'd prefer

Reply:
sg ty!